include/pybind11/pytypes.h

```diff
@@ -1824,8 +1824,8 @@ class capsule : public object {
         }
     }
 
-    capsule(const void *value, void (*destructor)(void *)) {
-        m_ptr = PyCapsule_New(const_cast<void *>(value), nullptr, [](PyObject *o) {
+    capsule(const void *value, void (*destructor)(void *), const char *name = nullptr) {
```

Comment:
note: this argument order is purposefully different from other ctors. Also currently the dtor does not free it, we may want to copy it and ensure we free it.